### DISCUSSION TITLE
Swiftfin build requirements?

### DISCUSSION THREAD

**Discussion by cooksey** [jellyfin/Swiftfin#1322] [Questions] [answered]
What are the requirements for building Swiftfin? The contributing file says Xcode 15, but it looks like the project actually requires Xcode 16. I'm running Xcode 15.2 (I need to update my laptop) and I'm getting failures from the Factory and CoreStore dependencies, and when I restricted them to earlier versions I'm getting errors related to icon assets and some SwiftUI code.

**Comment by LePips** [comment, marked as answer]
We do require Xcode 16 now, the documentation is out of date. You can update that, if you'd like.
